src/main/java/org/sonarqube/gradle/SonarPropertyComputer.java

```diff
@@ -136,8 +136,9 @@ private void computeDefaultProperties(Project project, Map<String, Object> prope
 
     overrideWithUserDefinedProperties(project, rawProperties);
 
-    // This is required if "sonar.sources" are neither found nor defined by user
+    // This is required if "sonar.sources" or "sonar.tests" are neither found nor defined by user; possible undesired behaviors due to the copy of those properties from the parent
     rawProperties.putIfAbsent(ScanProperties.PROJECT_SOURCE_DIRS, "");
+    rawProperties.putIfAbsent(ScanProperties.PROJECT_TEST_DIRS, "");
```

Comment:
Let's add a more specific command here
```suggestion
    // These empty assignments are required because modules with no `sonar.sources` or `sonar.tests` value inherit the value from their parent module.
    // This can eventually lead to a double indexing issue in the scanner-engine.
    rawProperties.putIfAbsent(ScanProperties.PROJECT_SOURCE_DIRS, "");
    rawProperties.putIfAbsent(ScanProperties.PROJECT_TEST_DIRS, "");
```

Reply:
We can link to this [ticket](https://sonarsource.atlassian.net/browse/SCANGRADLE-151) and add a TODO comment.